provider/server/machine_volume_detach_test.go

```diff
@@ -0,0 +1,146 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package server_test
+
+import (
+	"github.com/digitalocean/go-libvirt"
+	iri "github.com/ironcore-dev/ironcore/iri/apis/machine/v1alpha1"
+	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	libvirtutils "github.com/ironcore-dev/libvirt-provider/pkg/libvirt/utils"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"libvirt.org/go/libvirtxml"
+)
+
+var _ = Describe("DetachVolume", func() {
+	It("should correctly detach volume from machine", func(ctx SpecContext) {
+		By("creating a machine with two empty disks")
+		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
+			Machine: &iri.Machine{
+				Metadata: &irimeta.ObjectMetadata{
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: &iri.MachineSpec{
+					Power: iri.Power_POWER_ON,
+					Class: machineClassx3xlarge,
+					Volumes: []*iri.Volume{
+						{
+							Name: "disk-1",
+							EmptyDisk: &iri.EmptyDisk{
+								SizeBytes: 5368709120,
+							},
+							Device: "oda",
+						},
+						{
+							Name: "disk-2",
+							EmptyDisk: &iri.EmptyDisk{
+								SizeBytes: 5368709120,
+							},
+							Device: "odb",
+						},
+					},
+				},
+			},
+		})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(createResp).NotTo(BeNil())
+
+		DeferCleanup(func(ctx SpecContext) {
+			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
+				Expect(err).ShouldNot(HaveOccurred())
+				err = libvirtConn.DomainDestroy(libvirt.Domain{UUID: libvirtutils.UUIDStringToBytes(createResp.Machine.Metadata.Id)})
```

Comment:
why are you calling DomainDestroy?

Reply:
Add TODO comment, saying after garcefull shutdown this issue will be fixed

Reply:
Modified test to use boot image and so now its not needed